File: Ecommerce.Model/Entities/Product.cs
﻿namespace Ecommerce.Model.Entities
{
    public class Product : IEntityBase
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public string Size { get; set; }
        public string Color { get; set; }

        public int CategoryId { get; set; }
        public Category Categories { get; set; }
        public int Id { get; set; }
    }
}

File: Ecommerce.Web/ClientApp/core/domain/product.ts
export class Product {

    Id: number;
    CategoryId: number;
    Name: string;
    Description: string;
    Size: string;
    Color: string;

    constructor(id: number, categoryid: number, name: string, description: string, size: string, color: string) {
        this.Id = id;
        this.Name = name;
        this.CategoryId = categoryid;
        this.Description = description;
        this.Size = size;
        this.Color = color;
    }
}
